From 6ef6dd5927d34123bbd8fb8bede17a732fadb668 Mon Sep 17 00:00:00 2001
Message-Id: <6ef6dd5927d34123bbd8fb8bede17a732fadb668.1654590036.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Fri, 27 May 2022 16:38:14 +0200
Subject: [PATCH] Use /data as platform storage location

---
 src/platform/Linux/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/platform/Linux/BUILD.gn b/src/platform/Linux/BUILD.gn
index 7f46aee3dc..906e0b7557 100644
--- a/src/platform/Linux/BUILD.gn
+++ b/src/platform/Linux/BUILD.gn
@@ -33,6 +33,12 @@ if (chip_mdns == "platform") {
 }
 
 static_library("Linux") {
+  defines = [
+    "FATCONFDIR=\"/data\"",
+    "SYSCONFDIR=\"/data\"",
+    "LOCALSTATEDIR=\"/data\"",
+  ]
+
   sources = [
     "../DeviceSafeQueue.cpp",
     "../DeviceSafeQueue.h",
-- 
2.36.1

